### PR TITLE
Carousel basic logic

### PR DESCRIPTION
**Pull request by rvrita**
Carousel basic logic, styled component refactor, proptypes, tests is still TODO.

### CODE REVIEW

**Comment by ttreit** [comment]
Oops I hit approve before I left a general comment - looks great to me! I made a few notes that could be an edge case just so you can double check them and make sure they don't cause you any trouble down the road.

File: client/components/Carousel.jsx [hunk 1]
```diff
@@ -1,19 +1,142 @@
...
 class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstItemIndex: 0,
+      translate: 0,
+    };
+  }
+
+  handleLeftArrowClick() {
```
[the remark this answers is not in the document]
Comment (rvrita) [reply client/components/Carousel.jsx:104]: Not if you're using an arrow function, they are binding this to the object that defined them.
`() => this.handleLeftArrowClick()` in the onClick line 127 & 139.

File: client/components/Carousel.jsx [hunk 2]
```diff
@@ -1,19 +1,142 @@
...
+    this.setState((state) => ({
+      firstItemIndex: state.firstItemIndex - 4,
+      translate: state.translate + 828,
+    }), () => console.log(this.state));
+  }
+
+  handleRightArrowClick() {
```
[the remark this answers is not in the document]
Comment (rvrita) [reply client/components/Carousel.jsx:111]: Same comment as handleLeftArrow 😄

File: client/components/Carousel.jsx [hunk 3]
```diff
@@ -1,19 +1,142 @@
...
+    super(props);
+    this.state = {
+      firstItemIndex: 0,
+      translate: 0,
+    };
+  }
+
+  handleLeftArrowClick() {
+    this.setState((state) => ({
+      firstItemIndex: state.firstItemIndex - 4,
```
[the remark this answers is not in the document]
Comment (rvrita) [reply client/components/Carousel.jsx:106]: Yeah it's intentional. Right arrow should not show up if there is only 3 image left, you should not be able to scroll right.

File: client/components/Carousel.jsx [hunk 4]
```diff
@@ -1,19 +1,142 @@
...
-        <ul>
-          {props.allImages.map((image) => (
-            <li key={image}><img alt="review" src={image} /></li>
-          ))}
-        </ul>
-      </div>
+      <Container>
+        <Heading>Photos from reviews</Heading>
+        {/* Arrow conditional: */}
+        {state.firstItemIndex > 0
```
[the remark this answers is not in the document]
Comment (rvrita) [reply client/components/Carousel.jsx:125]: Yes, firstItemIndex can be 0, but if it is, we don't want the left arrow to show up, you should not be able to scroll left.

**Comment by rvrita** [comment]
Thanks for the review and double checking my edge cases! 👯
